library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java

```diff
@@ -118,27 +119,32 @@
      */
     private static final int PUBLIC_DOWNLOAD_ID = 6;
 
-    static {
-        sURIMatcher.addURI(AUTHORITY, "my_downloads", MY_DOWNLOADS);
-        sURIMatcher.addURI(AUTHORITY, "my_downloads/#", MY_DOWNLOADS_ID);
-        sURIMatcher.addURI(AUTHORITY, "all_downloads", ALL_DOWNLOADS);
-        sURIMatcher.addURI(AUTHORITY, "all_downloads/#", ALL_DOWNLOADS_ID);
-        sURIMatcher.addURI(AUTHORITY, "my_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
-        sURIMatcher.addURI(AUTHORITY, "all_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
-        // temporary, for backwards compatibility
-        sURIMatcher.addURI(AUTHORITY, "download", MY_DOWNLOADS);
-        sURIMatcher.addURI(AUTHORITY, "download/#", MY_DOWNLOADS_ID);
-        sURIMatcher.addURI(AUTHORITY, "download/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
-        sURIMatcher.addURI(AUTHORITY, Downloads.Impl.PUBLICLY_ACCESSIBLE_DOWNLOADS_URI_SEGMENT + "/#", PUBLIC_DOWNLOAD_ID);
+    public static void initURIMatcher(Context context) {
+        if (AUTHORITY == null) {
+            AUTHORITY = determineAuthority(context);
+            BASE_URIS = new Uri[]{
+                    Downloads.Impl.CONTENT_URI(AUTHORITY),
+                    Downloads.Impl.ALL_DOWNLOADS_CONTENT_URI(AUTHORITY),
+            };
+
+            sURIMatcher.addURI(AUTHORITY, "my_downloads", MY_DOWNLOADS);
+            sURIMatcher.addURI(AUTHORITY, "my_downloads/#", MY_DOWNLOADS_ID);
+            sURIMatcher.addURI(AUTHORITY, "all_downloads", ALL_DOWNLOADS);
+            sURIMatcher.addURI(AUTHORITY, "all_downloads/#", ALL_DOWNLOADS_ID);
+            sURIMatcher.addURI(AUTHORITY, "my_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
+            sURIMatcher.addURI(AUTHORITY, "all_downloads/#/" + Downloads.Impl.RequestHeaders.URI_SEGMENT, REQUEST_HEADERS_URI);
+            // temporary, for backwards compatibility
```

Comment:
yeah delete all that it uses I mean - simplify (as we do not have / need backwards compat)